### PR TITLE
Fixed the line length of the new Symfony Styles

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

**After/Before screenshots**

example_1

example_2

example_3

And now my question: is there any way to get the terminal dimensions from SymfonyStyles class without having to copy/paste the code found on \Console\Application?

### CODE REVIEW

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 1]
```diff
@@ -307,4 +310,83 @@ private function getProgressBar()
...
+            if (preg_match('/rows.(\d+);.columns.(\d+);/i', $sttyString, $matches)) {
+                return (int) $matches[2];
+            }
+            // extract [w, h] from "; h rows; w columns"
+            if (preg_match('/;.(\d+).rows;.(\d+).columns/i', $sttyString, $matches)) {
+                return (int) $matches[2];
+            }
+        }
+
+        return;
```
Comment (WouterJ) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: this has to be removed to comply with symfony CS

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 2]
```diff
@@ -307,4 +310,83 @@ private function getProgressBar()
...
+
+        $descriptorspec = array(1 => array('pipe', 'w'), 2 => array('pipe', 'w'));
+        $process = proc_open('mode CON', $descriptorspec, $pipes, null, null, array('suppress_errors' => true));
+        if (is_resource($process)) {
+            $info = stream_get_contents($pipes[1]);
+            fclose($pipes[1]);
+            fclose($pipes[2]);
+            proc_close($process);
+
+            if (preg_match('/--------+\r?\n.+?(\d+)\r?\n.+?(\d+)\r?\n/', $info, $matches)) {
```
Comment (WouterJ) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: why not replacing `--------` with `-{8}`?

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 3]
```diff
@@ -307,4 +310,83 @@ private function getProgressBar()
 
         return $this->progressBar;
     }
+
+    private function getTerminalWidth()
+    {
+        if ($this->terminalWidth) {
```
Comment (WouterJ) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: the property is never set atm

**Comment by stof** [comment]
@javiereguiluz the class should have a way to get them from the Application IMO

**Comment by javiereguiluz** [comment]
@stof I agree with you, but I don't know how to do it :cry:

**Comment by 1ed** [comment]
@javiereguiluz as a last resort [webmozart/console/ConsoleIO.php#L56-L58](https://github.com/webmozart/console/blob/d63a9a4c7ca806a1a7e81c2020b4791c98fcc146/src/IO/ConsoleIO.php#L56-L58)

**Comment by javiereguiluz** [comment]
@1ed that's a nice trick! Thanks for the reference. I've just updated the code as of 70a1f80

**Comment by javiereguiluz** [comment]
The errors reported by Travis seems to be caused by volatile tests (as reported by @stof in #14078):

```
1) Symfony\Component\Process\Tests\SigchildDisabledProcessTest::testIdleTimeoutNotExceededWhenOutputIsSent
Failed asserting that false is true.
/home/travis/build/symfony/symfony/src/Symfony/Component/Process/Tests/AbstractProcessTest.php:854

2) Symfony\Component\Process\Tests\SimpleProcessTest::testIdleTimeoutNotExceededWhenOutputIsSent
Failed asserting that false is true.
/home/travis/build/symfony/symfony/src/Symfony/Component/Process/Tests/AbstractProcessTest.php:854

3) Symfony\Component\Stopwatch\Tests\StopwatchTest::testStop
Failed asserting that 229 matches expected 200.
/home/travis/build/symfony/symfony/src/Symfony/Component/Stopwatch/Tests/StopwatchTest.php:79
```

**Comment by fabpot** [comment]
Thank you @javiereguiluz.

**Comment by ogizanagi** [comment]
@javiereguiluz : Now that this PR is merged and #14032 too, Travis reports the following error :

> 1) Symfony\Bundle\SecurityBundle\Tests\Functional\UserPasswordEncoderCommandTest::testEncodePasswordEmptySalt
str_repeat(): Second argument has to be greater than or equal to 0


I think this might be related to this PR.
See https://github.com/symfony/symfony/pull/14032#issuecomment-90508603

**Comment by 1ed** [comment]
See #14250

**Pull request closed without merging**